packages/core/data-transfer/src/strapi/remote/flows/index.ts

```diff
@@ -0,0 +1,78 @@
+import type { TransferStage } from '../../../../types';
+
+export type Step =
+  | { kind: 'action'; action: string }
+  | { kind: 'transfer'; stage: TransferStage; locked?: boolean };
+
+export { default as DEFAULT_TRANSFER_FLOW } from './default';
+
+interface IState {
+  step: Step | null;
+}
+
+export const createFlow = (flow: readonly Step[]) => {
+  const state: IState = { step: null };
+
+  /**
+   * Equality check between two steps
+   */
+  const stepEqual = (stepA: Step, stepB: Step): boolean => {
+    if (stepA.kind === 'action' && stepB.kind === 'action') {
+      return stepA.action === stepB.action;
+    }
+
+    if (stepA.kind === 'transfer' && stepB.kind === 'transfer') {
+      return stepA.stage === stepB.stage;
+    }
+
+    return false;
+  };
+
+  /**
+   * Find the index for a given step
+   */
+  const findStepIndex = (step: Step) => flow.findIndex((flowStep) => stepEqual(step, flowStep));
+
+  return {
+    has(step: Step) {
+      return findStepIndex(step) !== -1;
+    },
+
+    can(step: Step) {
+      if (state.step === null) {
+        return true;
+      }
+
+      const delta = findStepIndex(step) - findStepIndex(state.step);
```

Comment:
```suggestion
      const isSameStep = findStepIndex(step) - findStepIndex(state.step);
```
this might be a better way to name it, delta is a bit ambiguous

Reply:
It returns a number (difference between the two indexes) and not a boolean value, hence the name.
I can use smth like indexDifference or smth similar to make it clearer if you want?

Reply:
yeah stepsIndexDifference would be better, or you can turn the actual operation into a boolean value using "!" operator if you are not planning to use the actual number.

Reply:
[We do use the number value ](https://github.com/strapi/strapi/pull/15801/files/a19d73f2acdb7f7277c5f6203741e4382bc7dbad#diff-ffd0a0604e6e773d18b788cbfe6621d6fac993f52c95cf7ddea3a6535c1237d2R49-R53)

I'll change the variable name :slightly_smiling_face:

Reply:
Done